src/disco/topo/fd_topob.c

```diff
@@ -392,6 +393,95 @@ validate( fd_topo_t const * topo ) {
   }
 }
 
+void
+fd_topob_auto_layout( fd_topo_t * topo ) {
+  /* Incredibly simple automatic layout system for now ... just assign
+     tiles to CPU cores in NUMA sequential order, except for a few tiles
+     which should be floating. */
+
+  char const * FLOATING[] = {
+    "metric",
+    "bencho",
+    "bhole",  // FIREDANCER only
+  };
+
+  char const * ORDERED[] = {
+    "benchg",
+    "benchs",
+    "net",
+    "quic",
+    "verify",
+    "dedup",
+    "pack",
+    "bank",
+    "poh",
+    "pohi",   // FIREDANCER only
+    "shred",
+    "store",
+    "storei", // FIREDANCER only
+    "sign",
+    "gossip", // FIREDANCER only
+    "repair", // FIREDANCER only
+    "replay", // FIREDANCER only
+    "thread", // FIREDANCER only
+    "sender", // FIREDANCER only
+  };
+
+  for( ulong i=0UL; i<topo->tile_cnt; i++ ) {
+    fd_topo_tile_t * tile = &topo->tiles[ i ];
+    tile->cpu_idx = ULONG_MAX;
+  }
+
+  ulong cpu_ordering[ USHORT_MAX ] = { 0UL };
```

Comment:
Not a huge deal, but this is kind of big for the stack and to 0 like that. 512 kB. Is `FD_TILE_MAX` enough?

Reply:
This is number of cores on the machine, I guess it's probably fine.